[ISSUE-37] [ISSUE-41] Refactoring main menu to make it its own dialogue side sheet

Kayram2710 opened the pull request:
# Goal
The purpose behind this PR is to refactor the main menu into its own side sheet, this achieves the following:
1. Allows to easily have the menu connected to a page
2. Clears up the clutter on that page
3. By encorporating the Materials API it makes the app more usable

# Potential Issues:
- The materials API has limititation when it comes to setting the width of a modal side sheet, as a result the only solution i found was overwriting its variable by adding a custom width in the dimens.xml, this however will apply to any sheet.
- There is an issue where spam clicking the open menu opens up multiple instances at once, this type of issue is common all through out the app, it migth be worth setting a general buffer on event listeners

Kayram2710 committed:
Implemented Main Menu as Side Seet

Still need to fix a few bugs

Kayram2710 committed:
Make main menu start from the Left

Kayram2710 committed:
Fixed Width Bug

Kayram2710 committed:
Merge branch 'main' into BUG-37

Kayram2710 committed:
Reworked for main

damle22 committed:
E2E Test for MainMenuDialog

damle22 commented:
Since you made changes to the MainController by replacing it with  'MainMenuDialog', the unit tests that I wrote for the 'MainMenuController' are not working anymore. I removed them and created a E2E test that takes into account the new 'MainMenuDebug'. The end-to-end test is under 'MainMenuDialogE2ETest'.

damle22 commented on app/src/androidTest/java/minicap/concordia/campusnav/MainMenuDialogE2ETest.java:
This is the E2E test that I added

damle22 committed:
Fix ToggleCampus Button .setText bug

There was a minor error when someone tried to refractor the code. It is actually the other way around:         campusTextView.setText(showSGW ? "LOY" : "SGW");
If showSGW is true, it means we are on the SGW campus, therefore we want the text button to show 'LOY' so the user when he wants can toggle the button and go to LOY campus.

damle22 committed:
Added ToggleCampusE2ETest

Kayram2710 committed:
Merge branch 'main' into BUG-37

Kayram2710 committed:
Fixed merge conflict

Kayram2710 committed:
Addressing ISSUE 41

Kayram2710 committed:
Fixed Campus Map not opening

Kayram2710 committed:
Added bus Schedule open feature

Kayram2710 committed:
Added temporary placeholder for directions bottom fragment

Checking with Gabriel if I can go ahead and do it

Kayram2710 committed:
Added directions menu

Kayram2710 commented:
The main menu dialogue requires certain routines to be ran with slight differances between them, this is causing the sonarqube to detect duplicate code, will look into it.

Kayram2710 committed:
Modulated to minimize duplication

zachariyaJaveri reviewed (approved):
Everything has been addressed, great implementation!
